source/VersionHandlerImpl/src/ExecutionEnvironment.cs

```diff
@@ -25,26 +25,43 @@ namespace Google {
 /// </summary>
 internal class ExecutionEnvironment {
 
+    /// <summary>
+    /// Cached lower case CommandLine
+    /// </summary>
+    private static string Cached_Environment_CommandLine_Lower = null;
+    
+    /// <summary>
+    /// Get/Set cached lower case CommandLine
+    /// </summary>
+    public static string Environment_CommandLine_Lower {
+        get {
+            if (Cached_Environment_CommandLine_Lower == null) {
+                Cached_Environment_CommandLine_Lower = Environment.CommandLine.ToLower();
+            }
+            return Cached_Environment_CommandLine_Lower;
+        }
+    }
+    
     /// <summary>
     /// Whether the editor was started in batch mode.
     /// </summary>
     public static bool InBatchMode {
-        get { return Environment.CommandLine.ToLower().Contains("-batchmode"); }
+        get { return Environment_CommandLine_Lower.Contains("-batchmode"); }
```

Comment:
This is better than before but couldn't you just read all three of the switches it's interested in inside a constructor and just return booleans in the properties? This still runs unnecessary string operations every frame.